keras/layers/wrappers.py

```diff
@@ -122,7 +122,7 @@ def step(x, states):
             # to process batches of any size
             # we can go with reshape-based implementation for performance
             X = K.reshape(X, (-1, ) + input_shape[2:])  # (nb_samples * timesteps, ...)
-            y = self.layer.call(X)  # (nb_samples * timesteps, ...)
+            y = self.layer.call(X, mask)  # (nb_samples * timesteps, ...)
```

Comment:
Isn't that already handled by the rnn? 

Example case:
```python
xin = Input(batch_shape=(batch, time), dtype='int32')
xemb = Embedding(feat, embed)(xin)
xrnn = LSTM(rnn_size, return_sequences=True)(x_emb)
xout = TimeDistributed(Dense(feat, activation='softmax'))(xrnn)
```
- input shape: `(batch, time)`
- post embedded shape: `(batch, time, embed)`
- post lstm shape (with return sequences):` (batch, time, rnn_size)`
- then, one would want to apply a `TimeDistributed(Dense(...))` over this tensor. 

then

- The mask that came out of the embedding is 2dim (it's the same size as the input)
- The mask gets pushed through `xrnn` as is
- Then, the mask hits the `TimeDistributed`, of shape` (batch, time)`.  

next,

- the mask gets passed to `TimeDistributed`'s rnn, and reshaped to `(time, batch)`.  
- in the `theano_backend. rnn`, it does `output = T.switch(mask, output, output_tm1)`, 
- each step through the rnn is size `(batch,)`.  is this right? 
- I am assuming that this means the batch examples which were 0'ed out and the time steps which shouldn't be calculated are properly handled.  Is this right?

If I'm wrong on any part of that, please let me know.